Core/src/main/java/com/paypal/android/core/api/EligibilityAPI.kt

```diff
@@ -0,0 +1,44 @@
+package com.paypal.android.core.api
+
+import android.util.Log
+import com.paypal.android.core.CoreConfig
+import com.paypal.android.core.api.models.APIResult
+import com.paypal.android.core.api.models.Eligibility
+import com.paypal.android.core.graphql.common.GraphQlClient
+import com.paypal.android.core.graphql.fundingEligibility.FundingEligibilityQuery
+import com.paypal.android.core.graphql.fundingEligibility.models.FundingEligibilityIntent
+import com.paypal.android.core.graphql.fundingEligibility.models.SupportedCountryCurrencyType
+import com.paypal.android.core.graphql.fundingEligibility.models.SupportedPaymentMethodsType
+
+class EligibilityAPI(
+    private val graphQlClient: GraphQlClient,
+    private val coreConfig: CoreConfig
+) {
+
```

Comment:
The primary constructor could be internal for testing purposes, and then expose a public one that just takes the `CoreConfig`. I dont think we need to expose the `GraphQLClient` to the merchant. They dont need to now that its graphql.
```suggestion
class EligibilityAPI internal constructor (
    private val graphQlClient: GraphQlClient,
    private val coreConfig: CoreConfig
) {

    constructor(coreConfig: CoreConfig) : this(GraphQlClient(), coreConfig)
```